upgrader.go

```diff
@@ -70,120 +74,131 @@ func WithLogger(l log15.Logger) Option {
 // be writeable by the process using tableroll.
 // Canonically, this directory is `/run/${program}/tableroll/`.
 // Any number of options to configure tableroll may also be provided.
-func New(coordinationDir string, opts ...Option) (*Upgrader, error) {
-	return newUpgrader(realOS{}, coordinationDir, opts...)
+// If the passed in context is cancelled, the upgrader will no longer service
+// upgrades and will be in a 'Stopped' state.
+func New(ctx context.Context, coordinationDir string, opts ...Option) (*Upgrader, error) {
```

Comment:
Interesting the idea that the context allows you to cancel the whole state means one of two things:

1. The `Stop` method is now superfluous and can be removed
2. It's much harder to use the passed-in context as a way to time-out *just* the 'become owner' piece of functionality

I think we have a few options to resolve that:
1. The passed in context is only used for initial setup and we keep the Stop() method as the way to cancel serving upgrades
2. We pass in *two* contexts, one for setup and one for servicing upgrades and remove the Stop() method
3. We pass in a single context for New() which controls the full life-time of the upgrader object, and then we add a new method `BecomeOwner` or `Start` which you call afterwards that just encapsulates the piece of code to become the owner. it takes in a context which controls just the lifetime of that call. we remove the `Stop` method in favor of cancelling the context passed to `New`.

i think option 1 is probably most idiomatic?

Reply:
Agreed, 1 seems like the best option. I'll switch to that.